app/views/hyrax/file_sets/_actions.html.erb

```diff
@@ -1,57 +1,6 @@
-  <div class="btn-group">
-    <button class="btn btn-default dropdown-toggle" data-toggle="dropdown" type="button" id="dropdownMenu_<%= file_set.id %>" aria-haspopup="true" aria-expanded="false">
-      <span class="sr-only"><%= t('.press_to') %> </span>
-      <%= t('.header') %>
-      <span class="caret" aria-hidden="true"></span>
-    </button>
-
-    <ul role="menu" class="dropdown-menu dropdown-menu-right" aria-labelledby="dropdownMenu_<%= file_set.id %>">
-      <% if can?(:edit, file_set.id) %>
-        <li role="menuitem" tabindex="-1">
-          <%= link_to t('.edit'), edit_polymorphic_path([main_app, file_set]),
-            { title: t('.edit_title', file_set: file_set), id: "file_edit" } %>
-        </li>
-
-        <li role="menuitem" tabindex="-1">
-          <%= link_to t('.versions'),  edit_polymorphic_path([main_app, file_set], anchor: 'versioning_display'),
-            { title: t('.versions_title'), id: "file_versions" } %>
-        </li>
-      <% end %>
-
-      <% if can?(:destroy, file_set.id) %>
-        <li role="menuitem" tabindex="-1">
-          <%= link_to t('.delete'), polymorphic_path([main_app, file_set]),
-            method: :delete, 
-            title: t('.delete_title', file_set: file_set),
-            id: "file_destroy",
-            data: { confirm: t('.delete_confirm', file_set: file_set, application_name: application_name) } %>
-        </li>
-      <% end %>
-      
-      <% if can?(:download, file_set.id) %>
-        <li role="menuitem" tabindex="-1">
-          <%= link_to t('.download'),
-                      hyrax.download_path(file_set),
-                      title: t('.download_title', file_set: file_set),
-                      target: "_blank",
-                      id: "file_download",
-                      data: { label: file_set.id } %>
-        </li>
-
-        <% if file_set.pdf? %>
-          <li role="menuitem" tabindex="-1">
-            <%= link_to t('.read_and_annotate'), hyku_addons.pdf_viewer_path(hyrax.download_url(file_set.id).split('/')&.last&.split('?')&.first),  { title: 'Annotate', id: "file_read", target: "_blank" } %>
-          </li>
-        <% end %>
-      <% end %>
-    </ul>
-  </div>
-<% elsif can?(:download, file_set.id) %>
-  <%= link_to t('.download'),
-    hyrax.download_path(file_set),
-    title: t('.download_title', file_set: file_set),
-    target: "_blank",
-    id: "file_download",
-    data: { label: file_set.id } %>
+<% view_condition = [can?(:edit, file_set.id), can?(:destroy, file_set.id), can?(:download, file_set.id), (Flipflop.enabled?(:annotation) && file_set.pdf?)] %>
+<% if can?(:edit, file_set.id) || (view_condition.any? && !view_condition.one?)  %>
+  <%= render partial: 'hyrax/file_sets/actions_list', locals: {file_set: file_set} %>
+<% elsif view_condition.one? %>
+  <%= render partial: 'hyrax/file_sets/actions_button', locals: {file_set: file_set} %>
 <% end %>
```

Comment:
Im not sure about how this is written as im struggling a little how the condition will compute. So, any of the `view_condition`s are true but not just one of them? And the array of conitions contains `can?(:edit, file_set.id)` as well as the actual condition, which i assume is why you have the `!.one?`. 

I think its a smell that the array of conditions is 162 characters long. I would suggest its best to break up the different conditions into more manageable chunks and then the if condition will be far easier to comprehend.

```ruby
<% args = { file_set: file_set } %>
<% can_annotate = Flipflop.enabled?(:annotation) && file_set.pdf? %>
<% can_edit = can?(:edit, file_set.id) %>
<% can_maintain = can_edit && (can?(:destroy, file_set.id) || can?(:download, file_set.id)) %>

<% if can_maintain && can_annonate  %>
  <%= render partial: "hyrax/file_sets/actions_list", locals: args %>
<% elsif can_edit %>
  <%= render partial: "hyrax/file_sets/actions_button", locals: args %>
<% end %>
```

I'm not 100% sure if the `can_maintain` conditions match yours, its just an example of how i'd tackle a condition like this. I also wish we could use `?` in variable names as they'd be so much nicer.

Reply:
I think everything should be moved to a helper and we can just called **can_maintain && can_annonate ** which will be both helpers or what ever new name they are called.
Or better still have just one helper to call in the view.

Reply:
Helpers would require the fileset to be passed and also have access to the current request scope. I'm not sure we need helpers for a scenario where its only done once. If we needed to do this in a couple of places, then it would be best in a helper, but the `can?` method is already a helper.